src/main/java/io/github/moulberry/notenoughupdates/profileviewer/GuiProfileViewer.java

```diff
@@ -4937,6 +4941,60 @@ public String getSkinType() {
 				0
 			);
 		}
+
+		if (skillInfo != null) {
+			if(Constants.WEIGHT == null) {
+				Utils.showOutdatedRepoNotification();
+				return;
+			}
+
+			SenitherWeight senitherWeight = new SenitherWeight(skillInfo);
+			LilyWeight lilyWeight = new LilyWeight(skillInfo, profileInfo);
+
+			Utils.drawStringCentered(
+				EnumChatFormatting.GREEN + "Senither Weight: " + EnumChatFormatting.GOLD + numberFormat.format(roundToNearestInt(senitherWeight.getTotalWeight().getRaw())),
+				fr,
+				guiLeft + 63,
+				guiTop + 18,
+				true,
+				0
+			);
+
+			try {
+				int textWidth = fontRendererObj.getStringWidth("Senither Weight: " + numberFormat.format(roundToNearestInt(senitherWeight.getTotalWeight().getRaw())));
+				if (mouseX > guiLeft + 63 - textWidth / 2 &&
+					mouseX < guiLeft + 63 + textWidth / 2) {
+					if (mouseY > guiTop + 12 && mouseY < guiTop + 12 + fontRendererObj.FONT_HEIGHT) {
+						tooltipToDisplay = new ArrayList<>();
+						tooltipToDisplay.add(EnumChatFormatting.GREEN + "Skills: " + EnumChatFormatting.GOLD + numberFormat.format(roundToNearestInt(senitherWeight.getSkillsWeight().getWeightStruct().getRaw())));
+						tooltipToDisplay.add(EnumChatFormatting.GREEN + "Slayer: " + EnumChatFormatting.GOLD + numberFormat.format(roundToNearestInt(senitherWeight.getSlayerWeight().getWeightStruct().getRaw())));
+						tooltipToDisplay.add(EnumChatFormatting.GREEN + "Dungeons: " + EnumChatFormatting.GOLD + numberFormat.format(roundToNearestInt(senitherWeight.getDungeonsWeight().getWeightStruct().getRaw())));
+					}
+				}
+			} catch (Exception ignored) {}
+
+			Utils.drawStringCentered(
+				EnumChatFormatting.GREEN + "Lily Weight: " + EnumChatFormatting.GOLD + numberFormat.format(roundToNearestInt(lilyWeight.getTotalWeight().getRaw())),
+				fr,
+				guiLeft + 63,
+				guiTop + 28,
+				true,
+				0
+			);
+
+			try {
+				int fontWidth = fontRendererObj.getStringWidth("Lily Weight: " + numberFormat.format(roundToNearestInt(lilyWeight.getTotalWeight().getRaw())));
+				if (mouseX > guiLeft + 63 - fontWidth / 2 &&
+					mouseX < guiLeft + 63 + fontWidth / 2) {
+					if (mouseY > guiTop + 22 && mouseY < guiTop + 22 + fontRendererObj.FONT_HEIGHT) {
+						tooltipToDisplay = new ArrayList<>();
+						tooltipToDisplay.add(EnumChatFormatting.GREEN + "Skills: " + EnumChatFormatting.GOLD + numberFormat.format(roundToNearestInt(lilyWeight.getSkillsWeight().getWeightStruct().getRaw())));
+						tooltipToDisplay.add(EnumChatFormatting.GREEN + "Slayer: " + EnumChatFormatting.GOLD + numberFormat.format(roundToNearestInt(lilyWeight.getSlayerWeight().getWeightStruct().getRaw())));
+						tooltipToDisplay.add(EnumChatFormatting.GREEN + "Dungeons: " + EnumChatFormatting.GOLD + numberFormat.format(roundToNearestInt(lilyWeight.getDungeonsWeight().getWeightStruct().getRaw())));
+					}
+				}
+			} catch (Exception ignored) {}
+		}
 	}
```

Comment:
Although not strictly speaking a problem (or a doctrine we've followed till now), maybe move this to its own function. 500 lines is too much for a class, let alone a single function.

Reply:
oh and also `(Exception ignored)` is generally a bad idea. if you are anticipating an exception, try a close scope, if you want to be sure "just in case" and aren't sure what exceptions could be caused, please log the exception, or it will be impossible to find the cause of any bug in here.

Reply:
I had taken the tooltip code from the networth tooltip code which is why I had the try-catch. There doesn't seem to be any way for an exception to be thrown in the weight code so I've removed it.